src/particles/PhysicsParticles.hpp

```diff
@@ -642,44 +695,67 @@ template <typename problem_t> class PhysicsParticleRegister
 		return false;
 	}
 
-	// TODO(cch): replace name with particleType to get rid of the string comparison
+	// Utility method to convert particle type to string name (for writing plotfiles/checkpoints)
+	[[nodiscard]] static auto getParticleTypeName(ParticleType type) -> std::string
+	{
+		switch (type) {
+			case ParticleType::Rad:
+				return "Rad_particles";
+			case ParticleType::CIC:
+				return "CIC_particles";
+			case ParticleType::CICRad:
+				return "CICRad_particles";
+			default:
+				return "Unknown_particles";
```

Comment:
This should probably call `amrex::Abort` if it reaches this case.

Reply:
Can this be replaced by a member function of PhysicsParticleRegister that is specialized based on particle type?